apps/profiles/viewsets.py

```diff
@@ -308,6 +310,44 @@ def list_user_is_participant(self, request, user_id):
         serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
+    @action(
+        methods=["get"],
+        detail=True,
+        permission_classes=[IsAuthenticated],
+        url_path="events/favorite_events",
+        url_name="user_favorite_events",
+    )
+    def list_user_favorite_events(self, request, user_id):
+        queryset = self.get_queryset().filter(id__in=Subquery(
+            FavoriteEvent.objects.filter(user_id=user_id).values('event_id')))
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
```

Comment:
You need to check if requested user exists. Check, for example,  ```list_created_by_user``` function and find the way how to do it. Then add it to all of your endpoints here.

Reply:
Changes made

Reply:
Let's add these to get query set method:

```
                    Q(is_visible=True) & Q(type="open") |
                    Q(participants__in=[self.request.user.id]) & Q(type="private") |
                    Q(created_by=self.request.user) & Q(type="private")
```

Reply:
Check if user_id has indexes in both tables, add them if necessary for performance reasons

Reply:
Read and keep in mind N+1 issue for future